src/lib/form/range-slider/range-slider.component.ts

```diff
@@ -80,29 +93,51 @@ export class RangeSliderComponent implements OnInit, AfterViewInit, OnDestroy {
     }
 
     public inputChanged(event: Event): void {
-        const value = parseInt((event.target as HTMLInputElement).value) || 0;
+        this.setCurrentValues(parseInt((event.target as HTMLInputElement).value) || 0);
+    }
+
+    get formattedValue(): string {
+        return (this.symbol ? (this.symbol + ' ') : '') + (+this.value ?? '');
+    }
+
+    get formattedMin(): string {
+        return (this.symbol ?? '') + (+this.min);
+    }
+
+    get formattedMax(): string {
+        return (this.symbol ?? '') + (+this.max);
+    }
+
+    private setCurrentValue(value: number): void {
+        this.value = value;
+    }
+
+    private setCurrentValues(value: number, emitEvent = true) {
         const difference = this.max - this.min;
-        const percentage = Math.round((value / difference) * 100);
-        this.value.next([value, percentage]);
+        let mappedValue = Math.max(value, this.min) - this.min;
+        const percentage = Math.round((mappedValue / difference) * 100);
+        this.values.next([value, percentage, emitEvent]);
+        this.value = value;
         this.moveSlider(percentage);
     }
 
-    private changeValue(percentage: number): void {
-        const value = Math.round((percentage / 100) * this.max);
-        this.value.next([value, percentage]);
+    private changeValue(percentage: number, emitEvent = true): void {
+        const difference = this.max - this.min;
+        const value = Math.round((percentage / 100) * difference) + this.min;
+        this.values.next([value, percentage, emitEvent]);
         this.renderer.setProperty(this.input.nativeElement, 'value', value);
-        this.rawValue = value;
+        this.value = value;
     }
 
     private enableSlide(event: Event): void {
         this.windowListeners.forEach(eventEnder => eventEnder());
         this.windowListeners = [
             this.renderer.listen(this.window, 'mouseup', event => this.disableSlide(event)),
-            this.renderer.listen(this.window, 'touchend', event => this.disableSlide(event)),
+            // this.renderer.listen(this.window, 'touchend', event => this.disableSlide(event)),
```

Comment:
This listener is not needed?

Reply:
It might still be needed, but for now could be causing issues. It's one of the things I'll adjust later